### PR TITLE
Add ability to search with 'return'

### PR DESCRIPTION
**Pull request by s-espinosa**
@jdliss @jeneve Our work on issue 58 allowing users to use return to search.

# What does this add?

Allows user to use enter button to search instead of having to click on the search button.

# Requested feedback

* Should we be pushing the main.bundle.js and main.bundle.js.map files as well? Does that just clog the works unnecessarily?
* Is there any reason we should hesitate to change this to `onKeyUp` from `onChange`?

### CODE REVIEW

File: lib/components/SearchBarAndListings.js [hunk 1]
```diff
@@ -31,6 +31,7 @@ class SearchBarAndListings extends React.Component {
   }
 
   handleInputChange(event) {
+    if(event.keyCode === 13) { this.sendSearchQuery(); }
```
Comment (selfup) [inline lib/components/SearchBarAndListings.js]: If you are using a simple if statement that runs one line, it would be more "ES^" like to:

```javascript
  if (event.keyCode === 13) this.sendSearchQuery();
```

File: lib/components/SearchBarAndListings.js [hunk 2]
```diff
@@ -86,7 +87,7 @@ class SearchBarAndListings extends React.Component {
       <div>
         <div className="search-bar">
           <FormGroup>
-            <FormControl type="text" placeholder="Search jobs by location or technology" id="search-bar-input" onChange={this.handleInputChange.bind(this)}/>
+            <FormControl type="text" placeholder="Search jobs by location or technology" id="search-bar-input" onKeyUp={this.handleInputChange.bind(this)}/>
```
Comment (selfup) [inline lib/components/SearchBarAndListings.js]: Good catch from a typing standpoint.

I wonder how a paste event would be handled?

Like if someone were to paste in some text, would this search?

Just a thought!

**Comment by selfup** [comment]
### Hesitation concern

`onKeyUp` is great for typing. Most people type.

However, some people know they make mistakes. So they google a word and paste it in.

Would this prevent people from searching? As long as they are aware that they can click search after pasting, I am sure everything will be good to go.

Extreme edge case. But worth considering!

File: lib/components/SearchBarAndListings.js (hunk 2 above)
Comment (adamki) [inline lib/components/SearchBarAndListings.js]: I don't want to detract from any patterns that you've established here, just one point though.

using `.bind(this)` in the `render()` of your component is a known performance concern. Many are achieving the same binding by including the following in the `constructor` of the component:

```
constructor() {
    super();
    this.state = { jobs: {}, inputValue: "", activePage: 1, lastCall: "all" };

    this.onKeyUp = this.onKeyUp.bind(this);
}
```

This helps organize your component and provides a documentation-like structure. **caveat** this does affect React Hot-Loading

tl;dr : http://stackoverflow.com/a/31302493

Comment (jeneve) [inline lib/components/SearchBarAndListings.js]: Thanks for your suggestion @selfup, we checked it out and found that pasting code was no problem. We decided to keep out curly braces on our function since that style seems to be the standard in the project.

Comment (jeneve) [inline lib/components/SearchBarAndListings.js]: @adamki- wow that is interesting! we are opening a separate issue to research this further and consider implementing it across the repo. Will def. tag you in any PR's that result!

**Comment by selfup** [comment]
@jeneve Could you point me to a part of the project where a one line return uses curly braces?

Ok so I found a part that is:

```javascript
      if (searchInput === technologies[i]){
        techSearch = true;
      }
```

So I would follow that syntax instead.

```javascript
if (event.keyCode === 13) { 
  this.sendSearchQuery();
}
```

**Comment by s-espinosa** [comment]
@selfup Thanks for all the feedback! We looked into the question of pasting into the search box and found that when you pasted using right-click/control click that state wasn't updating to reflect the search term. Adjusted this so that state is set as it was before (onChange), but added an onKeyUp listener to check for 'enter'. Good catch!

**Comment by selfup** [comment]
So you are keeping the one line if statement even though the project uses 3 line if statements?

Just making sure!

File: lib/components/SearchBarAndListings.js [hunk 3]
```diff
@@ -30,6 +30,10 @@ class SearchBarAndListings extends React.Component {
     });
   }
 
+  checkReturn(event) {
+    if (event.keyCode === 13) { this.sendSearchQuery(); }
```
Comment (selfup) [inline lib/components/SearchBarAndListings.js]: Just making sure this does not need to be (following the project syntax):

```javascript
if (event.keyCode === 13) { 
  this.sendSearchQuery(); 
}
```

**Comment by jeneve** [comment]
I don't think we've decided. Either way I think we'll add an issue about
normalizing the syntax everywhere, as you suggested. I personally prefer it
on one line, but that is harder to make one rule about, since it depends on
the line length....

**Comment by selfup** [comment]
I like the thought process! Sounds good to me!

**Comment by s-espinosa** [comment]
@jeneve @jdliss Ran npm run build and pushed the new main.bundle.js. Recommend we merge and continue working other issues raised in this discussion on a new branch.

**Comment by jdliss** [comment]
@s-espinosa @jeneve I agree! I'll take a look at normalizing all the syntax stuff if that's a direction we want to go in.
